python/ray/data/datasource/file_based_datasource.py

```diff
@@ -136,10 +169,15 @@ class FileBasedDatasource(Datasource[Union[ArrowRow, Any]]):
     and tailored to particular file formats. Classes deriving from this class
     must implement _read_file().
 
+    If the _file_extension is defined, per default only files with this extension
+    will be read. If None, no default filter is used.
+
     Current subclasses:
         JSONDatasource, CSVDatasource, NumpyDatasource, BinaryDatasource
     """
 
+    _file_extension: Optional[Union[str, List[str]]] = None
```

Comment:
Nit: May want go with with an all-caps private attribute to make it clear that it's a static attribute, not an instance attribute?
```suggestion
    _FILE_EXTENSION = "csv"
```